crates/sargon/src/signing/collector/signatures_collector_preprocessor.rs

```diff
@@ -102,7 +103,7 @@ impl SignaturesCollectorPreprocessor {
                     EntitySecurityState::Securified { value } => {
                         let general_role =
                             GeneralRoleWithHierarchicalDeterministicFactorInstances::try_from(
-                                (value.security_structure.matrix_of_factors, RoleKind::Primary) // Should change according to kind
+                                (value.security_structure.matrix_of_factors, role_kind.clone())
```

Comment:
`RoleKind` should probably be `Copy` and then you can run ` cargo clippy --fix --allow-staged` which will remove this `.clone()`.

Reply:
😲

Reply:
yes I run `clippy --fix` before commiting, always :)